### DISCUSSION TITLE
Alerts and limits based on metrics (cost, evaluation, user-feedback), trigger webhook/email/slack/...

### DISCUSSION THREAD

**Discussion by marcklingen** [langfuse/langfuse#3997] [Ideas]
### Describe the feature or potential improvement

Suggested by @chrisshaw [here](https://github.com/orgs/langfuse/discussions/1512#discussioncomment-9757098), as alerts are very distinct compared to [subscribing to new events](https://github.com/orgs/langfuse/discussions/1033)

> Responding to events from individual scores, traces, etc seems fun but not urgent. In our case, we want to know that the moderation LLM app is still moderating correctly and want to be alerted based on an aggregation of scores crossing a threshold.
>
> Having looked at the linked concept, I wonder if this is really the answer. Not sure if this is what @hypernovas had in mind, but personally, what we're looking for is closer to getting alerted along a metric, not an event. Similar to hitting various thresholds in New Relic or Sentry.
> 
> As a concrete example, suppose we're using a large language model to look for abusive behavior. We want to know that it's continuing to perform its task adequately as models and behaviors drift. So we set up an evaluation (let's say a model-based eval or a custom eval) that runs on our generations. Every individual trace's score is going to be variable, but we want to know if something is seriously wrong with our LLM app if it falls beneath a certain average score, aggregated over the last hour. That's when we want to get alerted (whether it's slack or general webhook is less important).

**Comment by Kleyt0n** [comment]
I'm monitoring LLMs in production with Langfuse. It would be nice to have an alert feature (like a Slack connection) when the cost reaches a predefined limit.

**Comment by gsvn** [comment]
We have a similar setup & would be great to have alerts when a score goes below a certain average, aggregated over a configured period of time.

> So we set up an evaluation (let's say a model-based eval or a custom eval) that runs on our generations. Every individual trace's score is going to be variable, but we want to know if something is seriously wrong with our LLM app if it falls beneath a certain average score, aggregated over the last hour.

**Comment by maxdeichmann** [comment]
Hi all, we have a first webhook version out there: https://langfuse.com/docs/prompts/get-started#webhooks

If you have any questions or feedback, please let us know! Sending out webhooks on prompt versions is only the start. We will gradually increase the scope of the feature..

**Reply by aandis** [reply]
@maxdeichmann the link is invalid now. Do you have an up to date link for prompt version webhooks?

**Reply by aandis** [reply]
https://langfuse.com/docs/prompt-management/features/webhooks-slack-integrations

**Comment by joshwand** [comment]
This is critical for production monitoring, and is a blocker for adoption at one of my enterprise customers.

**Comment by marksalpeter** [comment]
Hi everyone! Our new [Monitors](https://cloud.langfuse.com/project/~/monitors) feature just shipped in cloud beta. You can set threshold alerts on metrics like cost, scores, and latency and route them to Slack, webhooks, or GitHub Actions ([docs](https://langfuse.com/docs/metrics/features/monitors)).

If you have any questions or feedback, please let us know.
cc: @Kleyt0n, @gsvn, @joshwand
